src/Symfony/Bundle/FrameworkBundle/Routing/Router.php

```diff
@@ -151,12 +152,16 @@ private function resolve($value)
             if (is_string($resolved) || is_numeric($resolved)) {
                 return (string) $resolved;
             }
+            if ($acceptArrayParameters && is_array($resolved)) {
+                return implode('|', $resolved);
+            }
 
             throw new RuntimeException(sprintf(
                 'The container parameter "%s", used in the route configuration value "%s", '.
-                'must be a string or numeric, but it is of type %s.',
+                'must be a %s, but it is of type %s.',
                 $match[1],
                 $value,
+                'string or numeric'.($acceptArrayParameters ? ' or an array' : ''),
```

Comment:
`'string '.($acceptArrayParameters ? ', an array ' : '').'or numeric'` to be grammatically correct :+1: